lib/raita-pipeline.ts

```diff
@@ -7,54 +7,137 @@ import {
+    const stackId =
+      envFromEnvironment === environments.prod
+        ? productionStackId
+        : getStackIdForNonProdEnvironment(branch);
+    return {
+      env: envFromEnvironment,
+      branch,
+      stackId,
+      authenticationToken: 'github-token',
+      tags: {
+        Environment: envFromEnvironment,
+        Project: 'raita',
+      },
+    };
+  }
+  throw new Error(
+    `Environment value ${envFromEnvironment} for ENVIRONMENT is not valid Raita environment.`,
+  );
+};
 
 /**
  * The stack that defines the application pipeline
  */
 export class RaitaPipelineStack extends Stack {
-  constructor(scope: Construct) {
-    const { config } = getPipelineConfig();
-    super(scope, 'raita-pipeline-' + config.env, {
-      env: {
-        region: config.region,
-      },
+  constructor(scope: Construct, id: string, props: StackProps) {
+    const config = getPipelineConfig();
+    super(scope, `stack-pipeline-raita-${config.stackId}`, {
+      ...props,
       tags: config.tags,
     });
 
-    const pipeline = new CodePipeline(this, 'Raita-Pipeline', {
-      pipelineName: 'Raita-' + config.env,
-      synth: new ShellStep('Synth', {
-        input: CodePipelineSource.gitHub(
-          'finnishtransportagency/raita',
-          config.branch,
-          {
-            authentication: SecretValue.secretsManager(
-              config.authenticationToken,
-            ),
-          },
-        ),
-        commands: [
-          'npm ci',
-          `npm run pipeline:synth --environment=${config.env} --branch=${config.branch}`,
-        ],
-      }),
-      dockerEnabledForSynth: true,
-      codeBuildDefaults: {
-        // TODO: Cacheing not working currently
-        cache: Cache.local(
-          LocalCacheMode.CUSTOM,
-          LocalCacheMode.SOURCE,
-          LocalCacheMode.DOCKER_LAYER,
-        ),
+    const pipeline = new CodePipeline(
+      this,
+      `pipeline-raita-${config.stackId}`,
+      {
+        pipelineName: `pl-raita-${config.stackId}`,
+        synth: new ShellStep('Synth', {
+          input: CodePipelineSource.gitHub(
+            'finnishtransportagency/raita',
+            config.branch,
+            {
+              authentication: SecretValue.secretsManager(
+                config.authenticationToken,
+              ),
+            },
+          ),
+          commands: [
+            'npm ci',
+            `npm run pipeline:synth --environment=${config.env} --branch=${config.branch} --stackid=${config.stackId}`,
+          ],
+        }),
+        dockerEnabledForSynth: true,
+        codeBuildDefaults: {
+          // TODO: Cacheing not working currently
+          cache: Cache.local(
+            LocalCacheMode.CUSTOM,
+            LocalCacheMode.SOURCE,
+            LocalCacheMode.DOCKER_LAYER,
+          ),
+        },
       },
-    });
-    pipeline.addStage(new RaitaApplication(this, config.env));
+    );
+    pipeline.addStage(
+      new RaitaApplicationStage(this, `raita`, {
```

Comment:
```suggestion
      new RaitaApplicationStage(this, `Raita`, {
```
This is just the name shown in CodePipelines. More consistent when capitalized. This was derived from env earlier, which forced the lowercasing. But using env didn't really make sense because we're not sharing pipelines, so the change to raita is definitely an improvement already!

Reply:
RaitaApplicationStage is now capitalized, name is 'Raita' as in suggestion. And true, this does not fall through to the stack resources as I thought.